esp/src/src/Session.ts

```diff
@@ -0,0 +1,60 @@
+import * as cookie from "dojo/cookie";
+import * as xhr from "dojo/request/xhr";
+import * as ESPUtil from "./ESPUtil";
+import * as topic from "dojo/topic"
+
+const espTimeoutSeconds = cookie("ESPSessionTimeoutSeconds") || 600; // 10 minuntes?
+const IDLE_TIMEOUT = espTimeoutSeconds * 10;
+const SESSION_RESET_FREQ = 30 * 1000;
+
+let _prevReset = Date.now();
+let idleWatcher = new ESPUtil.IdleWatcher(IDLE_TIMEOUT);
+let monitorLockClick = new ESPUtil.MonitorLockClick();
+let sessionIsActive = espTimeoutSeconds;
+
+declare const dojoConfig;
+
+export function initSession() {
+    if (sessionIsActive > -1) {
+        cookie("Status", "Unlocked");
+        cookie("ECLWatchUser", "true");
+
+        idleWatcher.on("active", function () {
+            _resetESPTime();
+        });
+        idleWatcher.on("idle", function (idleCreator) {
+            idleWatcher.stop();
+            topic.publish("hpcc/session_management_status", {
+                status: "Idle",
+                idleCreator: idleCreator
+            });
+        });
+
+        idleWatcher.start();
+        monitorLockClick.unlocked();
+    } else if (cookie("ECLWatchUser")) {
+        window.location.replace(dojoConfig.urlInfo.basePath + "/Login.html");
+    }
+}
+
+export function locked() {
+    idleWatcher.stop();
+}
+
+export function unlocked() {
+    idleWatcher.start();
+}
+
+export function fireIdle() {
+    idleWatcher.fireIdle();
+}
+
+export function _resetESPTime() {
```

Comment:
If its exported, then it should probably not have the "_" prefix?  If its only local to this file, then it should not be exported.

Reply:
Good point.